src/Compilers/CSharp/Test/Semantic/Semantics/AmbiguousOverrideTests.cs

```diff
@@ -1321,5 +1321,39 @@ public static void Main(string[] args)
             // since the runtime can distinguish signatures with different modopts.
             verifier.VerifyDiagnostics();
         }
+
+        [Fact]
+        [CompilerTrait(CompilerFeature.ReadonlyReferences)]
+        public void OverloadsWithDifferentParameterModifiers_Ref_RefReadOnly()
+        {
+            var text = @"
+abstract class TestClass
+{
+    public virtual void Method(ref int x) { }
+    public virtual void Method(ref readonly int x) { }
+}";
```

Comment:
Consider removing `abstract` and `virtual` so it's clear the overload error does not depend on virtual methods. Same comment for test below.